packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -647,3 +656,40 @@ function isMixedLogicalExpression(
 
   return false;
 }
+
+function isNodeOrNodeExpressionEqual(
+  a: TSESTree.Node,
+  b: TSESTree.Node,
+): boolean {
+  if (
+    a.type === AST_NODE_TYPES.ChainExpression ||
+    b.type === AST_NODE_TYPES.ChainExpression
+  ) {
+    if (a.type === AST_NODE_TYPES.ChainExpression) {
+      a = a.expression;
+    }
+    if (b.type === AST_NODE_TYPES.ChainExpression) {
+      b = b.expression;
+    }
```

Comment:
[Refactor] There's a `skipChainExpression` function in `prefer-promise-reject-errors`. I  tried reusing it here and all tests passed locally. `no-floating-promises`, `no-inferrable-types`, `prefer-find`, and `prefer-string-starts-ends-with` have some similar logics that I think could mostly or completely be switched to using the utility. WDYT?

This is definitely not a blocker, so if you don't see this by release time Monday, I can just merge and file a followup issue.